pkgs/hover/default.nix

```diff
@@ -1,81 +1,95 @@
-{ lib, buildGoModule, buildFHSUserEnv, pkgconfig, fetchFromGitHub,
-  stdenv, writeScript, xorg, libglvnd, addOpenGLRunpath, makeWrapper,
-  gcc, go, flutter }:
+{ lib
+, buildGoModule
+, buildFHSUserEnv
+, pkgconfig
+, fetchFromGitHub
+, stdenv
+, writeScript
+, xorg
+, libglvnd
+, addOpenGLRunpath
+, makeWrapper
+, gcc
+, go
+, flutter
+}:
 
 let
   pname = "hover";
   version = "0.43.0";
 
   libs = with xorg; [
-    libXi
-    libXxf86vm
-    libglvnd.dev
     libX11.dev
     libXcursor.dev
+    libXext.dev
+    libXi.dev
     libXinerama.dev
     libXrandr.dev
     libXrender.dev
+    libXfixes.dev
+    libXxf86vm
+    libglvnd.dev
     xorgproto
   ];
-in buildGoModule rec {
-  inherit pname version;
-
-  meta = with stdenv.lib; {
-    description = "A build tool to run Flutter applications on desktop";
-    homepage = "https://github.com/go-flutter-desktop/hover";
-    license = licenses.bsd3;
-    platforms = platforms.linux ++ platforms.darwin;
-  };
+  hover = buildGoModule rec {
+    inherit pname version;
 
-  subPackages = [ "." ];
+    meta = with stdenv.lib; {
+      description = "A build tool to run Flutter applications on desktop";
+      homepage = "https://github.com/go-flutter-desktop/hover";
+      license = licenses.bsd3;
+      platforms = platforms.linux ++ platforms.darwin;
+    };
 
-  vendorSha256 = "1wr08phjm87dxim47i8449rmq5wfscvjyz65g3lxmv468x209pam";
+    subPackages = [ "." ];
 
-  src = fetchFromGitHub {
-    rev = "v${version}";
-    owner = "go-flutter-desktop";
-    repo = pname;
-    sha256 = "0iw6sxg86wfdbihl2hxzn43ppdzl1p7g5b9wl8ac3xa9ix8759ax";
-  };
+    vendorSha256 = "1wr08phjm87dxim47i8449rmq5wfscvjyz65g3lxmv468x209pam";
+
+    src = fetchFromGitHub {
+      rev = "v${version}";
+      owner = "go-flutter-desktop";
+      repo = pname;
+      sha256 = "0iw6sxg86wfdbihl2hxzn43ppdzl1p7g5b9wl8ac3xa9ix8759ax";
+    };
+
+    nativeBuildInputs = [ addOpenGLRunpath makeWrapper ];
+
+    buildInputs = libs;
 
-  nativeBuildInputs = [ addOpenGLRunpath makeWrapper ];
+    checkRun = false;
 
-  propagatedBuildInputs = [
+    patches = [
+      ./fix-assets-path.patch
+    ];
+
+    postPatch = ''
+      sed -i 's|@assetsFolder@|'"''${out}/share/assets"'|g' internal/fileutils/assets.go
+    '';
+
+    postInstall = ''
+      mkdir -p $out/share
+      cp -r assets $out/share/assets
+      chmod -R a+rx $out/share/assets
+
+      wrapProgram "$out/bin/hover" \
+      --prefix LD_LIBRARY_PATH : ${stdenv.lib.makeLibraryPath libs}
```

Comment:
Compared to https://github.com/NixOS/nixpkgs/pull/80075#issuecomment-602215368 this is the most important change.

Since `hover` is being build inside a `buildGoModule` (instead of a `buildFHSUserEnv`) it still needs to be wrapped or it wouldn't found the X11 libraries that it links.